### PR TITLE
fix(cli): verbose and stats verbose flag (rebase)

### LINKED ISSUE #1443: Bug: webpack-cli --verbose throws TypeError: Cannot set property 'stats' of undefined
**Describe the bug**

**What is the current behavior?**
`webpack-cli --verbose` or `webpack-cli --verbose --stats verbose` throws an error described below.
```javascript
[webpack-cli] TypeError: Cannot set property 'stats' of undefined
    at StatsGroup.resolveOptions (/home/phoenix/Desktop/webpack-cli/packages/webpack-cli/lib/groups/StatsGroup.js:23:40)
    at StatsGroup.run (/home/phoenix/Desktop/webpack-cli/packages/webpack-cli/lib/groups/StatsGroup.js:33:14)
    at WebpackCLI._handleGroupHelper (/home/phoenix/Desktop/webpack-cli/packages/webpack-cli/lib/webpack-cli.js:220:40)
    at Promise.resolve.then.then.then.then.then.then.then (/home/phoenix/Desktop/webpack-cli/packages/webpack-cli/lib/webpack-cli.js:269:30)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:745:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
```

**To Reproduce**

Use either verbose flag or verbose flag with some value of stats flag. 

**Expected behavior**
Using verbose flags should output everything. If the verbose flag is used with stats flag then it should also output everything if the stats has a valid flag value else, it should also throw a warning if the stats flag value is not a valid one.

**Screenshots**
Screenshot from 2020-04-11 18-56-15

**Please paste the results of `webpack-cli info` here, and mention other relevant information**
```bash
System:
    OS: Linux 4.19 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (4) x64 Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
    Memory: 693.40 MB / 7.66 GB
  Binaries:
    Node: 10.15.2 - /usr/bin/node
    Yarn: 1.22.4 - /usr/bin/yarn
    npm: 5.8.0 - /usr/bin/npm
  Browsers:
    Chrome: 80.0.3987.149
    Firefox: 68.6.0esr
  npmPackages:
    webpack: ^4.42.0 => 5.0.0-beta.14 
    webpack-cli: ^3.3.11 => 4.0.0-beta.8 
```
**Additional context**

### PR DESCRIPTION
**Pull request by aman29271**
**What kind of change does this PR introduce?**
A bugfix which closes #1443 

**Did you add tests for your changes?**
Yes
**If relevant, did you update the documentation?**
Nope

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

File: test/info/info-help.test.js [hunk 1]
```diff
@@ -29,8 +29,8 @@ describe('should print help for info command', () => {
         expect(stderr).toHaveLength(0);
     });
 
-    it('should respect the --color=false flag', () => {
-        const { stdout, stderr } = runInfo(['info', 'help', '--color=false']);
+    it('should respect the color default true ', () => {
+        const { stdout, stderr } = runInfo(['info', 'help']);
```
Comment (evilebottnawi) [inline test/info/info-help.test.js:33]: Do not change existing tests if it is unnecessary

File: test/help/help-single-arg.test.js [hunk 2]
```diff
@@ -5,8 +5,8 @@ const { run } = require('../utils/test-utils');
 const helpHeader = 'The build tool for modern web applications';
 
 describe('single help flag', () => {
-    it('respects --color flag as false', () => {
-        const { stdout, stderr } = run(__dirname, ['--help', '--color=false'], false);
+    it('respects color default as true', () => {
+        const { stdout, stderr } = run(__dirname, ['--help'], false);
```
Comment (evilebottnawi) [inline test/help/help-single-arg.test.js:9]: Do not change existing tests if it is unnecessary

File: packages/webpack-cli/lib/groups/StatsGroup.js [hunk 3]
```diff
@@ -5,7 +5,9 @@ const logger = require('../utils/logger');
  */
 class StatsGroup extends GroupHelper {
     static validOptions() {
-        return ['none', 'errors-only', 'minimal', 'normal', 'detailed', 'verbose', 'errors-warnings'];
+        let validArrayString = ['none', 'errors-only', 'minimal', 'normal', 'detailed', 'verbose', 'errors-warnings'];
+        let validArrayObject = [{ verbose: true }, { verbose: true, stats: 'verbose' }];
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/groups/StatsGroup.js:9]: We can have more valid array when it is object

File: test/info/info-help.test.js (hunk 1 above)
Comment (aman29271) [reply test/info/info-help.test.js:33]: Okay.

Comment (aman29271) [reply test/info/info-help.test.js:33]: I am having trouble finding out `--color` flag on `v4`. It is in the WIP or something? That's why my test were failing locally. So, I have modified it little bit.

File: packages/webpack-cli/lib/groups/StatsGroup.js (hunk 3 above)
Comment (aman29271) [reply packages/webpack-cli/lib/groups/StatsGroup.js:9]: should i go for it? It would be more validated as well.

**Comment by aman29271** [comment]
This PR also closes #1475

File: packages/webpack-cli/lib/utils/GroupHelper.js [hunk 4]
```diff
@@ -22,14 +22,18 @@ class GroupHelper {
     }
 
     arrayToObject(arr) {
-        if (!arr) {
-            return;
+        if (Array.isArray(arr)) {
```
Comment (evenstensberg) [inline packages/webpack-cli/lib/utils/GroupHelper.js:25]: if(Array.isArray(arr) return 
if !arr return

Comment (aman29271) [reply packages/webpack-cli/lib/utils/GroupHelper.js:25]: I have added  object validation in tests. So, to verify  that if it's an `object` or an `Array`, I added that blocks.

Comment (evenstensberg) [reply packages/webpack-cli/lib/utils/GroupHelper.js:25]: That’s not what I’m saying

Comment (aman29271) [reply packages/webpack-cli/lib/utils/GroupHelper.js:25]: Thanks

Comment (aman29271) [reply packages/webpack-cli/lib/utils/GroupHelper.js:25]: I resolved it.

File: packages/webpack-cli/lib/groups/StatsGroup.js [hunk 5]
```diff
@@ -5,7 +5,12 @@ const logger = require('../utils/logger');
  */
 class StatsGroup extends GroupHelper {
     static validOptions() {
-        return ['none', 'errors-only', 'minimal', 'normal', 'detailed', 'verbose', 'errors-warnings'];
+        let validArrayString = ['none', 'errors-only', 'minimal', 'normal', 'detailed', 'verbose', 'errors-warnings'];
```
Comment (evenstensberg) [inline packages/webpack-cli/lib/groups/StatsGroup.js:8]: const

**Comment by snitin315** [comment]
suppressed by https://github.com/webpack/webpack-cli/pull/1549. hence closing, Thanks.

**Pull request closed without merging**